log4j-api-test/src/test/java/org/apache/logging/log4j/message/ParameterFormatterTest.java

```diff
@@ -17,156 +17,167 @@
+                        3,
+                        "Test message ab c"
+                },
+                new Object[]{
+                        "Test message {} {} {} {} {} {}",
+                        new Object[]{"a", null, "c", null, null, null},
+                        6,
+                        "Test message a null c null null null"
+                },
+                new Object[]{
+                        "Test message {}{} {}",
+                        new Object[]{"a", "b", "c", "unnecessary", "superfluous"},
+                        5,
+                        "Test message ab c"
+                },
+                new Object[]{
+                        "Test message \\{}{} {}",
+                        new Object[]{"a", "b", "c"},
+                        3,
+                        "Test message {}a b"
+                },
+                new Object[]{
+                        "Test message {}{} {}\\",
+                        new Object[]{"a", "b", "c"},
+                        3,
+                        "Test message ab c\\"
+                },
+                new Object[]{
+                        "Test message {}{} {}\\\\",
+                        new Object[]{"a", "b", "c"},
+                        3,
+                        "Test message ab c\\"
+                },
+                new Object[]{
+                        "Test message \\\\{}{} {}",
+                        new Object[]{"a", "b", "c"},
+                        3,
+                        "Test message \\ab c"
+                },
+                new Object[]{
+                        "Test message {}{} {}",
+                        new Object[]{"a", "b", "c"},
+                        3,
+                        "Test message ab c"
+                },
+                new Object[]{
+                        "Test message {} {} {} {} {} {}",
+                        new Object[]{"a", null, "c", null, null, null},
+                        6,
+                        "Test message a null c null null null"
+                },
+                new Object[]{
+                        "Test message {}{} {}",
+                        new Object[]{"a", "b", "c", "unnecessary", "superfluous"},
+                        5,
+                        "Test message ab c"
+                },
+                new Object[]{
+                        "Test message \\{}{} {}",
+                        new Object[]{"a", "b", "c"},
+                        3,
+                        "Test message {}a b"
+                },
+                new Object[]{
+                        "Test message {}{} {}\\",
+                        new Object[]{"a", "b", "c"},
+                        3,
+                        "Test message ab c\\"
+                },
+                new Object[]{
+                        "Test message {}{} {}\\\\",
+                        new Object[]{"a", "b", "c"},
+                        3,
+                        "Test message ab c\\"
```

Comment:
@rgoers, similar to the above, this was returning `Test message ab c\\\\` earlier.

Reply:
It has been two weeks. I am resolving this conversation.